File: Data Structures & Algorithms/counting-bits/submission-2.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        retList = [0] * (n + 1)

        for i in range(n + 1):
            temp = i

            while temp:
                temp &= temp - 1
                retList[i] += 1
        
        return retList

File: Data Structures & Algorithms/reverse-a-linked-list/submission-2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* node = head;
        ListNode* prevNode = nullptr;
        ListNode* temp = head->next;

        while (node) {
            node->next = prevNode;
            prevNode = node;
            node = temp;
            temp = node->next;
        }

        return prevNode;
    }
};
